lib/src/router/coordinator.dart

```diff
@@ -1,19 +1,52 @@
 import 'package:flutter/material.dart';
 import 'package:get_it/get_it.dart';
-import 'package:myapp/src/router/auto_router.dart';
-import 'package:myapp/src/router/auto_router.gr.dart';
+import 'package:go_router/go_router.dart';
+import 'package:myapp/src/router/route_name.dart';
+import 'package:myapp/src/router/router.dart';
 
 class XCoordinator {
   static XRouter get rootRouter => GetIt.I<XRouter>();
+  static final navigatorKey = GlobalKey<NavigatorState>();
+  static final shellKey = GlobalKey<NavigatorState>();
+  static BuildContext get context => navigatorKey.currentState!.context;
 
-  static BuildContext get context =>
-      rootRouter.navigatorKey.currentState!.context;
+  static void pop<T extends Object?>([T? result]) => context.pop(result);
 
-  static Future<bool> pop<T extends Object?>([T? result]) async {
-    return rootRouter.pop(result);
-  }
+  static void goNamed(
+    String name, {
+    Map<String, String> params = const <String, String>{},
+    Map<String, dynamic> queryParams = const <String, dynamic>{},
+    Object? extra,
+  }) =>
+      context.goNamed(
+        name,
+        params: params,
+        queryParams: queryParams,
+        extra: extra,
+      );
 
-  static Future showAuthentication() async {
-    return rootRouter.push(const SignRoute());
-  }
+  static void showHomeScreen() => context.goNamed(XRouteNames.home.name);
+  static void showAccountScreen() => context.goNamed(XRouteNames.account.name);
+
+  static Future<T?> showSignInScreen<T extends Object?>() =>
```

Comment:
We should design Navigation stack. and it is root stack.
the stack may like:
```
- Loading,
- Authentication,
                  - Signin
                  - Signup
                  - ForgotPassword
- Dashboard
                  - Home
                  - Profile
                  - Sample
                  - SampleDetail
```
Show for each stack/feature, It has its own Coordinator. for example
lib/src/router/coordinator.dart
 - showDashboard
 - showAuthentication
 
lib/src/feature/Authentication/router/auth_coordinator.dart
 - showSignup
 - showForgotPassword


lib/src/feature/dashboard/router/dashboard_coordinator.dart
 - showSample
 - showSampleDetail

Reply:
In go_router we can design the navigation stack by defining a route same level or inside another route to depict how the navigation stack will be implemented, you can check this in [router.dart](https://github.com/GoldenOwlAsia/mobile-flutter-template/blob/4c2914d3870e79692bb5bebcfef5ea8755afd889/lib/src/router/router.dart).
Is that what you mean? Or you need to split `coordinator.dart` into multiple files for each feature?